CMakeLists.txt

```diff
@@ -13,20 +13,33 @@
 # You should have received a copy of the GNU Lesser General Public License along
 # with the SymCC runtime. If not, see <https://www.gnu.org/licenses/>.
 
-cmake_minimum_required(VERSION 3.5)
-project(SymRuntime)
+cmake_minimum_required(VERSION 3.22)
+project(SymccRuntime)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 \
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
+
+set(CMAKE_CXX_STANDARD 17)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} \
 -Wredundant-decls -Wcast-align \
 -Wmissing-include-dirs -Wswitch-default \
 -Wextra -Wall -Winvalid-pch -Wredundant-decls -Wformat=2 \
 -Wmissing-format-attribute -Wformat-nonliteral")
 
-option(QSYM_BACKEND "Use the Qsym backend instead of our own" OFF)
+set(SYMCC_RT_AVAILABLE_BACKENDS "simple" "qsym")
+string(REPLACE ";" ", " SYMCC_RT_AVAILABLE_BACKENDS_FMT "\{${SYMCC_RT_AVAILABLE_BACKENDS}\}")
+
+set(SYMCC_RT_BACKEND "qsym" CACHE STRING "SymCC Runtime Backend to build.\
```

Comment:
We should make sure to update the documentation, mentioning that the QSYM backend is now the default. It makes sense, but we should communicate it.

Is there a reason why you're defining the variable with `set` instead of `option`? The latter should make it visible as a user setting in CMake UIs...

Reply:
Oh, I missed that `option` doesn't work for anything but Booleans! Then let's keep your `set` approach :+1: